src/Distribution/Nixpkgs/PackageMap.hs

```diff
@@ -7,45 +7,77 @@ import Control.Lens
 import qualified Data.Aeson as JSON
 import qualified Data.ByteString.Lazy as LBS
 import Data.Function
-import Data.List as List
-import Data.List.Split
+import Data.List (minimumBy)
 import Data.Map.Strict ( Map )
 import qualified Data.Map.Strict as Map
 import Data.Maybe
 import Data.Set ( Set )
 import qualified Data.Set as Set
 import Language.Nix
+import Paths_distribution_nixpkgs (getDataFileName)
 import System.Process
 
 type PackageMap = Map Identifier (Set Path)
 
-readNixpkgPackageMap :: [String] -> IO PackageMap
-readNixpkgPackageMap = fmap identifierSet2PackageMap . readNixpkgSet
+-- | Evaluate nixpkgs at a given (nix) path and build a 'Map'
+--   keeping track of all 'Path's that end in a given 'Identifier'
+--   and evaluate to a derivation.
+--   This can be used to find an attribute 'Path' for an arbitrary
+--   package name using 'resolve'.
+--
+--   Note: Evaluation of nixpkgs is very expensive (takes multiple
+--   seconds), so cache the result of this function if possible.
+--
+--   >>> readNixpkgPackageMap "<nixpkgs>" (Just "{ config = { allowAliases = false; }; }")
+--   fromList [ … ]
+readNixpkgPackageMap :: String
+                     -- ^ Path to nixpkgs, must be a valid nix path
+                     --   (absolute, relative or @NIX_PATH@ lookup)
+                     -> Maybe String
+                     -- ^ (Optional) argument attribute set to pass to
+                     --   nixpkgs. Must be a valid nix attribute set.
+                     -> IO PackageMap
+readNixpkgPackageMap = fmap (fmap identifierSet2PackageMap) . readNixpkgSet
```

Comment:
```suggestion
readNixpkgPackageMap nixpkgsPath nixpkgsArgs = do
  identSet <- readNixpkgSet nixpkgsPath nixpkgsArgs
  pure $ identifierSet2PackageMap identSet
```

Non point-free is probably easier to read here at a glance.

Reply:
Done.